### PR TITLE
build(solecs): clean up deps

### PR DESCRIPTION
**Pull request by holic**
related to https://github.com/latticexyz/mud/issues/411

I did this by going through each dep, looking at import usages, and deciding if it should be a regular dep (if used within the app bundle) or dev dep (if only used in tests, type checking, etc.). I also removed a couple extraneous deps.

### CODE REVIEW

File: packages/solecs/package.json [hunk 1]
```diff
@@ -21,32 +21,23 @@
...
-    "@solidstate/contracts": "^0.0.52",
-    "@typechain/ethers-v5": "^10.2.0",
-    "@types/mocha": "^9.1.1",
     "ds-test": "https://github.com/dapphub/ds-test.git#c9ce3f25bde29fc5eb9901842bf02850dfd2d084",
     "ethers": "^5.7.2",
+    "memmove": "https://github.com/dk1a/memmove.git#ffd71cd77b1708574ef46a667b23ca3a5cc9fa27",
+    "solmate": "https://github.com/transmissions11/solmate.git#9cf1428245074e39090dceacb0c28b1f684f584c"
+  },
+  "devDependencies": {
+    "@solidstate/contracts": "^0.0.52",
```
Comment (alvrs) [inline packages/solecs/package.json:33]: I think `@solidstate/contracts` should be a dependency so it's automatically installed by consumers

File: packages/solecs/package.json [hunk 2]
```diff
@@ -21,32 +21,23 @@
...
     "typechain": "^8.1.1",
     "typedoc": "0.23.21"
   },
-  "peerDependencies": {
-    "@ethersproject/abi": "^5.7.0",
-    "@ethersproject/providers": "^5.7.2",
-    "@solidstate/contracts": "^0.0.52",
-    "ds-test": "https://github.com/dapphub/ds-test.git#c9ce3f25bde29fc5eb9901842bf02850dfd2d084",
-    "ethers": "^5.7.2",
-    "forge-std": "https://github.com/foundry-rs/forge-std.git#b4f121555729b3afb3c5ffccb62ff4b6e2818fd3",
```
Comment (alvrs) [inline packages/solecs/package.json:48]: moving `forge-std` to dependencies so it's installed automatically by consumers

File: packages/solecs/package.json (hunk 1 above)
Comment (holic) [reply packages/solecs/package.json]: oops, good catch

File: packages/solecs/package.json (hunk 2 above)
Comment (holic) [reply packages/solecs/package.json:48]: just a dev dep should be fine for forge-std

**Pull request merged**